### DISCUSSION TITLE
Feature Request: Implement Neutrino Client-Side Block Filtering for Enhanced Privacy and Compatibility with Modern Bitcoin Core Versions

### DISCUSSION THREAD

**Discussion by warioishere** [bisq-network/bisq#6966] [Ideas]
Dear Bisq Developers,

I am writing to propose the integration of Neutrino Client-Side Block Filtering (as outlined in BIP157 and BIP158) into the Bisq network. This suggestion is prompted by the current dependency of Bisq on peer bloom filters and the evolving nature of Bitcoin Core, which now by default disables peer bloom filters in its newer versions.

**Background:**
Bisq currently operates with peer bloom filters enabled. However, recent versions of Bitcoin Core have peer bloom filters disabled by default, leading to compatibility issues. Users frequently need to make adjustments to get Bisq running with the latest Bitcoin Core versions, which is not an optimal user experience.

**Proposed Feature:**
Implementing Neutrino Client-Side Block Filtering offers a modern, privacy-focused alternative. This protocol allows lightweight clients to verify blockchain transactions with enhanced privacy and reduced network load, without revealing the addresses or transaction patterns to network nodes.

**Advantages:**
1. **Improved Privacy and Efficiency**: Neutrino enhances user privacy and network efficiency, addressing the limitations of bloom filters.
2. **Compatibility with Modern Bitcoin Core**: As Bitcoin Core evolves, it's crucial for Bisq to stay compatible. Adopting Neutrino ensures compatibility with the latest Bitcoin Core versions that have peer bloom filters disabled by default.
3. **User Experience**: This change would streamline the setup process for new and existing users, removing the need for additional configuration to ensure compatibility with Bitcoin Core.

**Conclusion:**
The integration of Neutrino Client-Side Block Filtering would not only enhance privacy and efficiency but also ensure Bisq's compatibility with the latest developments in Bitcoin technology. This aligns with the principles of security, privacy, and user-friendliness that are core to Bisq's philosophy.

Thank you for considering this feature request. Implementing this feature would be a significant step forward for Bisq and its community, ensuring the platform remains at the forefront of decentralized trading technology.

Best regards,
wario

**Comment by ghost** [comment]
Bisq relies on the BitcoinJ library, here is its status on Neutrino support: https://github.com/bitcoinj/bitcoinj/issues/1945
